test/unit/tracks.blackbox.test.ts

```diff
@@ -0,0 +1,182 @@
+import { describe, it, expect, vi, beforeEach } from 'vitest';
+import { fetchTracks, createTrack } from '../../src/services/api/tracks';
+import axiosInstance from '../../src/services/api/axios/config';
+import {
+  TrackSchema,
+  TrackResponseSchema,
+} from '../../src/schemas/trackSchemas';
+
+// Mock axios instance
+vi.mock('../../src/services/api/axios/config', () => ({
+  default: {
+    get: vi.fn(),
+    post: vi.fn(),
+  },
+}));
+
+// Mock TrackSchema
+vi.mock('../../src/schemas/trackSchemas', () => ({
+  TrackSchema: {
+    safeParse: vi.fn(),
+  },
+  TrackResponseSchema: {
+    safeParse: vi.fn(),
+  },
+}));
+
+describe('Tracks Service - Black Box Tests', () => {
+  beforeEach(() => {
+    vi.clearAllMocks();
+  });
+
+  describe('fetchTracks', () => {
+    it('should fetch tracks with pagination and filters', async () => {
+      // Arrange
+      const mockResponse = {
+        data: {
+          data: [
+            {
+              id: '1',
+              title: 'Test Track',
+              artist: 'Test Artist',
+              album: 'Test Album',
+              genres: ['Rock'],
+              coverImage: 'test.jpg',
+            },
+          ],
+          meta: {
+            totalPages: 5,
+            page: 1,
+          },
+        },
+      };
+
+      (axiosInstance.get as any).mockResolvedValue(mockResponse);
+      (TrackResponseSchema.safeParse as any).mockReturnValue({
+        success: true,
+        data: mockResponse.data,
+      });
+
+      // Act
+      const result = await fetchTracks(
+        1,
+        10,
+        'title',
+        { genre: 'Rock' },
+        'test'
+      );
+
+      // Assert - тестуємо тільки зовнішню поведінку
+      expect(result.isOk()).toBe(true);
+      if (result.isOk()) {
```

Comment:
Don't think this if statement is needed, if result.isOk is falsy the test is going to fail anyway